### ISSUE TITLE
[Ch 12][WikiDocs][Major] 부록 §6 "5가지 lever 카드" 참조가 페이지에서 죽은 참조 (Colab 에는 존재)

### ISSUE THREAD

**Issue by sy2399** [yoon-gu/neuqes-101#66]
## 대상

- **챕터**: Ch 12 — BERT Multi-class (Yelp 5클래스)
- **산출물**: WikiDocs (`pages/12-bert_multiclass-variation.md`, `pages/12-bert_multiclass-data_scaling.md`)
- **기준 커밋**: `eed3dbe` (페이지 최종 변경 `7ee7d0c`)
- **환경**: 정적 검토 — `pages/` ↔ 부록 노트북 대조

## 실제 결과

`pages/12-bert_multiclass-variation.md:140` 이 챕터 결론을 이렇게 닫습니다.

> 데이터를 더 부으면 BERT 가 더 벌리지만, 정확도를 크게 끌어올리려면 데이터 외에 *더 강한 모델·HPO·LLM distill* 같은 다른 lever 도 필요합니다 — **부록 §6 의 5가지 lever 카드 참조**.

그런데 부록 WikiDocs 페이지 `pages/12-bert_multiclass-data_scaling.md` (46줄)에 **lever 카드가 없습니다.** 페이지 구성이 다음으로 끝납니다.

```
## 12-3. 부록 — 데이터 스케일링 곡선
## 질문
## 실험 조건
## 결과
## 결과 해석
## 본편과의 연결
```

절 번호도 없어서 "§6" 이 가리킬 대상 자체가 없습니다.

## 내용은 Colab 에 있습니다 — 페이지로만 옮겨지지 않았습니다

부록 **노트북**(`12_bert_multiclass/12_bert_multiclass_data_scaling.ipynb`) 마지막 셀에는 해당 내용이 정상적으로 존재합니다.

> `## 6. 보너스 — BERT 성능을 끌어올리는 5가지 lever`
>
> | lever | 방법 | 대략 효과 |
> |---|---|---|
> | 1. 데이터 (양 + 질) | train size↑, 도메인 적응 사전학습(DAPT), 라벨 노이즈 정리 | +3-15%p |
> | 2. 더 강한 모델 | DistilBERT → RoBERTa-large / DeBERTa-v3-large | +2-5%p |
> | 3. HPO | learning rate · warmup · weight decay 튜닝 | +1-3%p |
> | 4. 학습 전략 | layer-wise LR decay(LLRD), 앙상블 | 각 +1-2%p |
> | 5. LLM 라벨링 + distill | 큰 LLM 으로 라벨 생성 → 작은 모델 증류 | +3-10%p |

따라서 **Colab 으로 부록을 여는 독자는 정상**이고, **WikiDocs·EPUB 독자만 빈 참조를 만납니다.** 원고 내용이 틀린 것이 아니라 노트북 → 페이지 이관에서 이 절이 빠진 건입니다.

## 기대 결과

둘 중 하나로 정리되어야 합니다.

- **(a)** 부록 페이지에 lever 절을 추가한다 (권장 — 본편이 이 내용을 결론의 다음 단계로 안내하고 있음)
- **(b)** 본편 `variation.md:140` 의 참조 문장에서 "부록 §6 의 5가지 lever 카드 참조" 를 제거한다

(a) 로 갈 경우, 부록 페이지에는 절 번호가 없으므로 참조 문구도 "§6" 대신 절 제목으로 바꾸는 편이 안전합니다.

## 재현 절차

```bash
# 본편 페이지의 참조
grep -n "lever" pages/12-bert_multiclass-variation.md

# 부록 페이지에 lever 절이 있는지
grep -n "lever\|보너스\|HPO\|distill" pages/12-bert_multiclass-data_scaling.md   # → 결과 없음

# 부록 노트북에는 있는지
grep -n "5가지 lever" 12_bert_multiclass/12_bert_multiclass_data_scaling.ipynb   # → 존재
```

WikiDocs 에서는 12-3(변형) 페이지 맨 아래 문장을 읽고 부록 페이지로 이동해 보면 재현됩니다.

## 추가 확인

- 가이드 3쪽 분류상 **"변환이 흘린 것 — 노트북에는 있는데 페이지에 없다"** 와 **"교차참조 어긋남"** 이 겹친 형태입니다.
- Ch 12 페이지의 **다른 출력 누락은 없습니다.** 본편 노트북 기준 출력 있는 셀 16개 ↔ 페이지 `▶ 실행 결과` 블록 16개로 완전히 일치하고, Trainer 학습 표도 텍스트로 정상 변환되어 있습니다.
- E1~E9 린터(`check_wikidocs_md.py`)는 Ch 12 6개 파일 모두 위반 0건입니다. 이 결함은 린터가 잡을 수 없는 종류입니다.

## 수정 범위 원칙

가이드 3쪽에 따라 이 수정은 **산문 범위**입니다. 코드와 `▶ 실행 결과` 는 `executed/` 가 단일 출처이므로 페이지에서 직접 손대지 않습니다. lever 표는 실행 결과가 아닌 설명 산문이므로 페이지에 추가해도 이 원칙에 어긋나지 않습니다. **노트북 재실행은 불필요**합니다.

## 전파 범위

- `book/chapters/ch12_bert_multiclass.tex` — 같은 참조 문장이 있고 부록 절에도 lever 표가 없어 EPUB 에서 동일하게 재현됩니다.

## 심각도

**Major** — 내용이 전달되긴 하나 독자가 안내받은 자료에 도달하지 못합니다. 실행 실패나 오학습이 아니므로 Blocker 는 아닙니다.

## 담당

- 수정 담당자: 미정
- 재검수자: 등록자 (@sy2399)
- 연결: 챕터 검수 이슈 #36

**Comment by sy2399** [comment]
## (a) 안으로 수정했습니다 — 부록 페이지에 lever 절 추가

브랜치 `fix/ch12-review-2026-09` · 커밋 `1825cf5`

`pages/12-bert_multiclass-data_scaling.md` 에 **`## 보너스 — BERT 성능을 끌어올리는 5가지 lever`** 절을 추가했습니다. 부록 노트북 §6 의 표 5행(데이터 / 더 강한 모델 / HPO / 학습 전략 / LLM distill)을 그대로 옮겼고, 마지막 문단의 "위 §4 곡선" 은 페이지에 절 번호가 없으므로 "위 결과 곡선" 으로 바꿨습니다.

### 참조 문구도 함께 고쳤습니다

이슈의 기대 결과에 적어 둔 대로, 부록 페이지에는 절 번호가 없어 "§6" 이 가리킬 대상이 없었습니다. 참조를 **절 제목 기준**으로 바꿔 노트북·페이지 양쪽에서 유효하게 만들었습니다.

```diff
- 부록 §6 의 5가지 lever 카드 참조.
+ 부록의 **보너스 — BERT 성능을 끌어올리는 5가지 lever** 절 참조.
```

노트북(`12_bert_multiclass.ipynb` §6)과 `pages/12-bert_multiclass-variation.md` 양쪽에 같은 문구를 적용했습니다. 노트북 쪽도 바꾼 이유는, 노트북 문구를 그대로 두면 **재변환 때 "§6" 이 다시 페이지로 흘러들어와 같은 결함이 재발**하기 때문입니다.

### 함께 넣은 것

`## 본편과의 연결` 절에 본편 §5(20K + bigram)와 부록(`TfidfVectorizer()` 기본값)의 sklearn 설정이 다르다는 단서를 추가했습니다. #59 의 (b) 안 수정과 짝을 맞춘 것으로, 독자가 두 숫자를 같은 곡선 위의 점으로 오해하지 않게 하기 위함입니다.

### 검증

`check_wikidocs_md.py` 전체 159개 파일 E1~E9 위반 0건입니다. 표 추가로 인한 규칙 위반은 없습니다.

`book/chapters/ch12_bert_multiclass.tex` 는 재빌드 대상입니다. 노트북 재실행은 불필요합니다.

_(2026-08-21 갱신: 병합 후 최종 검토 단계에서 세 축 확인을 마치고 종결했습니다. 교차 확인 대기 문구는 걷어냅니다.)_

**Comment by sy2399** [comment]
## 정정 — 이 결함은 WikiDocs 전용입니다 (EPUB 영향 없음)

이슈 본문의 **전파 범위** 를 정정합니다. "`book/chapters/ch12_bert_multiclass.tex` 에도 lever 표가 없어 EPUB 에서 동일하게 재현됩니다" 라고 적었는데 **사실이 아닙니다.** tex 에는 lever 표가 정상적으로 들어 있습니다.

```
book/chapters/ch12_bert_multiclass.tex:1803
  \subsection{보너스 --- BERT 성능을 끌어올리는 5가지 lever}
  \begin{booktable}{12장 보너스 - BERT 성능을 끌어올리는 5가지 lever 표}{tab:ch12-11}
```

tex 는 `book/tools/notebook_to_tex.py` 가 **노트북에서** 생성하고, 부록 노트북 §6 이 정상이므로 tex 도 정상입니다. 누락된 것은 손으로 쓴 WikiDocs 부록 페이지뿐이었습니다.

따라서 정확한 범위는 다음과 같습니다.

| 산출물 | 상태 |
|---|---|
| 부록 노트북 (Colab) | ✅ 정상 — §6 존재 |
| `book/chapters/*.tex` → EPUB | ✅ 정상 — `\subsection{보너스 ...}` 존재 |
| WikiDocs `pages/…-data_scaling.md` | ❌ 누락 → 이번에 수정 |

**EPUB 재빌드는 이 건 때문에는 필요하지 않습니다.** (단, 참조 문구를 절 제목 기준으로 바꾼 노트북 수정이 tex 에 반영되려면 `notebook_to_tex.py` 재생성이 필요한데, 그건 #59·#65 수정과 함께 도는 경로입니다. tex:1131 의 "부록 §6 의 5가지 lever 카드 참조" 는 tex 에서도 §6 번호가 없어 — 부록은 `\section{부록: …}` 아래 `\subsection` 으로 매겨집니다 — 절 제목 기준 문구가 tex 에서도 더 정확합니다.)

혼란을 드려 죄송합니다. `format:epub` 라벨은 붙이지 않았으니 라벨 조정은 필요 없습니다.

**Comment by jwrim9305** [comment]
## 검증했습니다 — 수정 확인, 누락 없음

`fix/ch12-review-2026-09` · `1825cf5` 기준으로 대조했습니다.

### ① 부록 페이지 lever 절

`pages/12-bert_multiclass-data_scaling.md:49` 에 `## 보너스 — BERT 성능을 끌어올리는 5가지 lever` 가 추가되었고, 표 5행이 부록 노트북 §6 과 일치합니다 (데이터 / 더 강한 모델 / HPO / 학습 전략 / LLM distill — 효과 수치까지 동일).

### ② 참조 문구

```diff
- 부록 §6 의 5가지 lever 카드 참조
+ 부록의 **보너스 — BERT 성능을 끌어올리는 5가지 lever** 절 참조
```

| 파일 | |
|---|---|
| `12_bert_multiclass.ipynb` (§6) | ✅ |
| `pages/12-bert_multiclass-variation.md` | ✅ |

**절 번호 대신 제목으로 가리키게 바꾼 판단이 좋았습니다.** 번호 체계가 없는 WikiDocs 페이지에서도 유효해졌고, 부록 노트북은 `## 6. 보너스 — …` 로 번호와 제목을 둘 다 갖고 있어 Colab 에서도 그대로 동작합니다.

## 한 곳 — tex 는 재생성 대기

`book/chapters/ch12_bert_multiclass.tex` 는 브랜치에서도 옛 문구(`부록 §6 의 5가지 lever 카드 참조`)입니다. tex 에는 `\subsection{보너스 --- BERT 성능을 끌어올리는 5가지 lever}`(`:1803`) 로 **번호 없는 절**이 들어 있어 "§6" 이 EPUB 에서도 가리킬 대상이 없습니다.

다만 누락이 아니라 순서 문제입니다 — tex 는 `book/tools/notebook_to_tex.py` 가 노트북에서 생성하므로, 노트북이 고쳐진 지금 **재생성하면 따라옵니다.** 두 번째 댓글에서 짚으신 대로입니다.

| 항목 | 상태 |
|---|---|
| 부록 페이지 lever 절 | ✅ |
| 노트북·페이지 참조 문구 | ✅ |
| tex 참조 문구 | ⬜ tex 재생성 시 반영 |

재생성까지 확인되면 이 이슈는 정리된 것으로 봅니다.

**Comment by sy2399** [comment]
## 재검토 — @sy2399 (슬롯 b2), 2026-08-21

**판정: 세 축 모두 해결. 닫습니다.**

- 수정 커밋: `1825cf5`(부록 페이지에 절 신설) + `fd9d661`(tex 참조 문구) — 통합 브랜치 `fix/ch12-review-2026-09-merged`
- 세 축 현재 상태
  - 📓 **Notebook** ✅ — 부록 `12_bert_multiclass_data_scaling.ipynb` 에 `## 6. 보너스 — BERT 성능을 끌어올리는 5가지 lever` 실재 (원래도 있었음)
  - 📄 **WikiDocs** ✅
    - `pages/12-bert_multiclass-data_scaling.md:51` 에 `## 보너스 — BERT 성능을 끌어올리는 5가지 lever` 절 신설 — **참조 대상이 실재하게 됨**
    - 참조하는 쪽 `pages/12-bert_multiclass-variation.md` 도 절 번호(§6)가 아니라 **이름 기반 참조**로 변경
  - 📕 **EPUB** ✅ — `fd9d661` 로 `ch12_bert_multiclass.tex:702` 수정
    - **추가로 발견한 것**: compact 원고 전환(master `f3453ab`) 이후 책은 부록을 `\onlineAppendixLink` 로 빼서 **책 안에 §6 이라는 절 자체가 없습니다.** 즉 EPUB 에서도 같은 참조가 죽어 있었습니다. 이름 기반 참조로 바꿔 세 축 모두 해소했습니다

**절 번호 참조는 산출물마다 번호 체계가 달라 잘 깨집니다.** 이번 건은 이름 기반 참조로 통일했습니다.

EPUB(pandoc 3.10.1) · PDF(XeTeX, 801쪽) 양쪽 빌드로 렌더링까지 확인했습니다. 재실행·재변환 불필요.
